docs/source/guides/cache_system.mdx

```diff
@@ -12,52 +12,51 @@ specific language governing permissions and limitations under the License.
 
 # Neuron Model Cache
 
-The Neuron Model Cache is a remote cache for compiled Neuron models in the `neff` format. 
-It is integrated into the [`NeuronTrainer`] class to enable loading pretrained models from the cache instead of compiling them locally. 
-This can speed up the training process by about –3x.
+The Neuron Model Cache is a remote cache for compiled Neuron models in the `neff` format.
+It is integrated into the [`NeuronTrainer` and `NeuronModelForCausalLM] classes to enable loading pretrained models from the cache instead of compiling them locally.
 
-The Neuron Model Cache is hosted on the [Hugging Face Hub](https://huggingface.co/aws-neuron/optimum-neuron-cache) and includes compiled files for all popular and supported pre-trained models `optimum-neuron`.
+The Neuron Model Cache is hosted on the [Hugging Face Hub](https://huggingface.co/aws-neuron/optimum-neuron-cache) and includes compiled files for all popular and supported `optimum-neuron` pre-trained models.
 
-When training a Transformers or Diffusion model with vanilla [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx), the models needs to be first compiled. The compiled version is stored in a local directory, usually `/var/tmp/neuron-compile-cache`. 
-This means that every time you train a new model in a new environment, you need to recompile it, which takes a lot of time. 
+When loading a Transformers or Diffusion model, it needs to be compiled to neuron format with [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx),
+in order to run on Neuron platforms.
+The compilation produces several compilation files stored in a local directory, usually `/var/tmp/neuron-compile-cache`.
+This means that every time you train or export a model on a new host, you need to recompile it, which takes a lot of time.
 
 We created the Neuron Model Cache to solve this limitation by providing a public cache of precompiled available models and a private cache to create your private, secured, remote model cache.
 
-The Neuron Model Cache plugs into the local cache directory of the Hugging Face Hub. During training, the [`NeuronTrainer`] will check if compilation files are available on the Hub and download them if they are found, allowing you to save both time and cost by skipping the compilation phase.
-
 ## How the caching system works
 
 ### Hash computation
 
-Many factors can trigger compilation among which: 
+Many factors can trigger compilation among which:
 
-- The model weights
-- The input shapes
-- The precision of the model, full-precision or bf16
-- The version of the Neuron X compiler
-- The number of Neuron cores used
+- The input shapes,
+- The precision of the model, full-precision or bf16,
+- The version of the Neuron X compiler,
+- The number of Neuron cores used,
+- Subtle modifications in the model graph.
```

Comment:
I would stress more on the fact that even a small configuration change will lead to a re-compilation.

Reply:
Done